### PR TITLE
[AI POC] Bot AI Virtual Assistant Implementation

### PR DESCRIPTION
**Pull request by ChetanSharma-msft**

### CODE REVIEW

**Review by Pawank-MSFT — COMMENTED** [review]
some comments please check

**Review by Wajeed-msft — APPROVED** [review]

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "name": "botvirtualassistant",
+    "version": "1.0.0",
+    "msteams": {
+        "teamsAppId": null
+    },
+    "description": "Microsoft Teams Toolkit AI Chat Bot Sample with Teams AI Library",
```
Comment (Pawank-MSFT) [inline samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json:7]: Update this description.

File: samples/bot-virtual-assistant/ai-skill-bot/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "ai-skill-bot",
+    "version": "1.0.0",
+    "description": "Bot Builder v4 echo skill bot sample",
```
Comment (Pawank-MSFT) [inline samples/bot-virtual-assistant/ai-skill-bot/package.json:4]: Please make sure this is correct?

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        "url": "https://github.com"
+    },
+    "dependencies": {
+        "@azure/openai": "^1.0.0-beta.12",
+        "@microsoft/teams-ai": "^1.1.0",
+        "applicationinsights": "^3.2.1",
+        "botbuilder": "^4.20.0",
+        "botframework-connector": "~4.22.1",
+        "dotenv": "~8.2.0",
+        "natural": "^7.0.6",
```
Comment (Pawank-MSFT) [inline samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json:34]: Please make sure if we using all the libraries, like "natural".

**Review by Pawank-MSFT — COMMENTED** [review]
some more comments

File: README.md [hunk 4]
```diff
@@ -40,6 +40,12 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |:--:|:-------------------|:---------------------------------------------------------------------------------------------------------------------------|:--------|:-------------|:-------------|
 |1|Hello World            | Microsoft Teams hello world sample app.                                           |[View][app-hello-world#cs]     |[View][app-hello-world#ts] ![toolkit-icon](assets/toolkit-icon.png) |[View](/samples/app-hello-world/csharp/demo-manifest/app-hello-world.zip)
 
+## [AI samples](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/?view=doc-intel-4.0.0)
+
+|    | Sample Name        | Description                                                                                                                | C#    | JavaScript | App Manifest  
+|:--:|:-------------------|:---------------------------------------------------------------------------------------------------------------------------|:--------|:-------------|:-------------|
+|1|bot-virtual-assistant|This is a sample application that demonstrates a root bot that routes inputs to skill bots for tailored responses.||[View][bot-virtual-assistant#nodejs]|-
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply README.md]: Anshul mention that manifest file is not required for this sample.
Even our resources are not working like blob storage, CosmosDB and Anshul will delete those resources later.

File: samples/bot-virtual-assistant/ai-skill-bot/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+- Microsoft Teams is installed and you have an valid [M365 account](https://www.microsoft.com/en-us/microsoft-365/what-is-microsoft-365?msockid=00bf51c23fe96f65186b42723e726e7f)
+- [NodeJS](https://nodejs.org/en/)
+- [dev tunnel](https://learn.microsoft.com/en-us/azure/developer/dev-tunnels/get-started?tabs=windows) or [ngrok](https://ngrok.com/) latest version or equivalent tunnelling solution
+- [Visual Studio Code](https://code.visualstudio.com/Download) or any other code editor
+- An [Azure subscription](https://learn.microsoft.com/en-us/microsoft-365/enterprise/subscriptions-licenses-accounts-and-tenants-for-microsoft-cloud-offerings?view=o365-worldwide) with permissions to create resources
+- [Azure Open AI Service](https://learn.microsoft.com/en-us/azure/ai-services/openai/overview)
+
+## Setup for App Registration
+
+1. Register a new application in the [Microsoft Entra ID � App Registrations](https://go.microsoft.com/fwlink/?linkid=2083908) portal.
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/README.md:40]: These are basic skill bots, so I think, not required for now.

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/.localConfigs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+BOT_ID=
+BOT_PASSWORD=
+SECRET_AZURE_OPENAI_API_KEY=
+AZURE_OPENAI_ENDPOINT=https://{servicename}.openai.azure.com/
+AZURE_OPENAI_DEPLOYMENT_NAME=
+MicrosoftAppType=Multitenant
+MicrosoftAppTenantId=
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/bot-ai-virtual-assistant/.localConfigs:7]: We have already mentioned in the Readme about keeping this as empty to support multi tenant.

File: samples/bot-virtual-assistant/ai-skill-bot/.env [hunk 7]
```diff
@@ -0,0 +1,8 @@
+MicrosoftAppType=
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/.env:1]: Yes, added one top readme file for this.

File: README.md [hunk 8]
```diff
@@ -70,7 +76,6 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |25| Tab External Auth | This sample illustrates how to implement Google auth using external auth providers. |[View][tab-external-auth#csharp]||
 |26| Tab Meeting Tanscript Recording |This sample shows meeting transcripts and recordings. |[View][tab-meeting-transcript-recording#csharp]|[View][tab-meeting-transcript-recording#nodejs] ![toolkit-icon](assets/toolkit-icon.png)|
 |27| Tab DeepLink Pass Values |Microsoft Teams sample which displays deeplinks based on the values passed dynamically to the context. ||[View][tab-deeplink-pass-values#ts]|
-|28| Nested App Authentication |This sample shows nested app authentication (NAA) is a new protocol that simplifies single sign-on (SSO) for single page applications. ||[View][tab-nested-auth#nodejs] ![toolkit-icon](assets/toolkit-icon.png)|
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply README.md:79]: I took readme file before merging auth sample.
I have added this now.

File: README.md [hunk 9]
```diff
@@ -282,7 +287,6 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 [tab-meeting-transcript-recording#nodejs]:samples/tab-meeting-transcript-recording/nodejs
 [tab-meeting-transcript-recording#csharp]:samples/tab-meeting-transcript-recording/csharp
 [tab-deeplink-pass-values#ts]:samples/tab-deeplink-pass-values/ts
-[tab-nested-auth#nodejs]:samples/tab-nested-auth/nodejs
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply README.md:292]: I took readme file before merging auth sample.
I have added this now.

File: samples/bot-virtual-assistant/ai-skill-bot/appPackage/manifest.json [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    "short": "AI bot that translates text",
+    "full": "An AI skill bot designed to translate user-provided text inputs and return the translated responses."
+  },
+  "accentColor": "#FFFFFF",
+  "bots": [
+    {
+      "botId": "{{BotId}}",
+      "scopes": [
+        "personal",
+        "team",
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/appPackage/manifest.json:31]: Yes, we can use but removing it.

File: samples/bot-virtual-assistant/ai-skill-bot/assets/sample.json [hunk 11]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-bot-virtual-assistant-ai-skill-bot-nodejs",
+    "source": "officeDev",
+    "title": "AI Translator Bot Sample",
+    "shortDescription": "This sample bot demonstrates the text translation using an AI-powered bot.",
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/assets/sample.json:6]: Updated this.

File: samples/bot-virtual-assistant/ai-skill-bot/bot.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            new AzureKeyCredential(process.env.AzureOpenAIApiKey)
+        );
+
+        // Event handler for incoming messages
+        this.onMessage(async (context, next) => {
+            // Prepare the messages array with system and user messages
+            const messages = [
+                {
+                    role: 'system',
+                    content: 'You are an AI assistant. Your task is to eliminate user instructions from query and translate the following text as per user inputs. Also, do not add any greetings or extra sentence in result'
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/bot.js:26]: This bot we have specifically created for translation only, so no need to update the prompt for this.

File: samples/bot-virtual-assistant/ai-skill-bot/README.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+If you want to deploy the app to Azure, you can follow the below steps:
+- Create an Azure app service in Azure portal that support `NodeJS`.
+- Deploy the code to the app service.
+
+To learn more about deploying a `NodeJS` app to Azure, see [Configure the App Service app and deploy the code](https://learn.microsoft.com/en-us/azure/app-service/quickstart-nodejs?tabs=windows&pivots=development-environment-vscode#configure-the-app-service-app-and-deploy-code) for a complete list of deployment instructions.
+
+## Further reading
+
+- [Implement a skill](https://learn.microsoft.com/en-us/azure/bot-service/skill-implement-skill?view=azure-bot-service-4.0&tabs=cs)
+- [Bot Basics](https://docs.microsoft.com/azure/bot-service/bot-builder-basics?view=azure-bot-service-4.0)
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/README.md:160]: Added one link for Azure OpenAI service.

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/env/.env.dev [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+# Updating AZURE_SUBSCRIPTION_ID or AZURE_RESOURCE_GROUP_NAME after provision may also require an update to RESOURCE_SUFFIX, because some services require a globally unique name across subscriptions/resource groups.
+AZURE_SUBSCRIPTION_ID=
+AZURE_RESOURCE_GROUP_NAME=
+RESOURCE_SUFFIX=
+
+# Generated during provision, you can also add your own variables.
+BOT_ID=
+TEAMS_APP_ID=
+BOT_AZURE_APP_SERVICE_RESOURCE_ID=
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/bot-ai-virtual-assistant/env/.env.dev:15]: Not needed here.
We are using .localConfigs

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/env/.env.local [hunk 15]
```diff
@@ -0,0 +1,13 @@
+# This file includes environment variables that can be committed to git. It's gitignored by default because it represents your local development environment.
+
+# Built-in environment variables
+TEAMSFX_ENV=local
+APP_NAME_SUFFIX=local
+
+# Generated during provision, you can also add your own variables.
+BOT_ID=
+TEAMS_APP_ID=
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/bot-ai-virtual-assistant/env/.env.local:9]: Not needed here.
We are using .localConfigs

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json (hunk 1 above)
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json:7]: Updated this.

File: samples/bot-virtual-assistant/ai-skill-bot/package.json (hunk 2 above)
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/ai-skill-bot/package.json:4]: Updated it.

File: samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json (hunk 3 above)
Comment (ChetanSharma-msft) [reply samples/bot-virtual-assistant/bot-ai-virtual-assistant/package.json:34]: Removed it.

**Review by Harikrishnan-MSFT — APPROVED** [review]
Looks good Approved!

**Pull request merged**